pages/your_tasks/readable_code.md

```diff
@@ -0,0 +1,99 @@
+---
+title: How to make your code readable
+description: Good practices to make your code readable
+contributors: ["Aleksandra Nenadic"]
+page_id: readable_code
+related_pages: []
+---
+
+
+## Why is it important to make your code readable?
+
+**Code readability** contributes to the *accessibility* and *reusability* of your code (the "A" and "R" in the [FAIR reseach software principles][fair-rs-principles][^1]) - 
+once people freely, legally and easily obtain a copy your software - they need to be able to read and understand it to reuse and extend it. 
+So, readable code helps create more reusable scientific software and can empower collaboration between researchers.
+
+In order to develop readable code, we should ask ourselves: "If I re-read this piece of code in fifteen days or one year, will I be able to understand 
+what I have done and why?" 
+Or: "If a new person who has just joined the project reads my code, will they be able to understand what I have written?"
+
+In addition, software requirements will likely change over time because software is dynamic in nature. 
+When these requirements change, a developer (who is not necessarily the same person who wrote the original code) needs to make changes to the software
+to satisfy these new and emerging needs. 
+They do this by reading the original code to understand it and identify what needs to change. 
+Readable code facilitates the the process of understanding the code, the evolution of the codebase, and saves future developers' time and effort.
+
+### Considerations
+
+- Readable code is easier to understand, maintain, debug and extend (reuse) - saving time and effort.
+- Readable code can also make it better and more secure in a way, as it is easier for reviewers to pick out errors.
+
+
+## What practices are there to help me improve the readability of my code?
+
+One of the most important things we can do to make sure our code is readable by others (and ourselves a few months down the line) is to make sure that it is descriptive, 
+cleanly and consistently formatted and uses sensible, descriptive names for variable, function and module names. 
+In order to help us format our code, we generally follow guidelines known as a style guide. 
+A style guide is a set of conventions that we agree upon with our colleagues or community, to ensure that everyone contributing to the same project is producing code which 
+looks similar in style. 
+While a group of developers may choose to write and agree upon a new style guide unique to each project, in practice many programming languages have a single style guide 
```

Comment:
Might be worth adding an example of such a style guide - would PEP8 for Python count here or would a project example be better?

Reply:
Done